.github/workflows/test-repro.yml

```diff
@@ -87,14 +97,16 @@ jobs:
           # after this step.
           set +e
 
-          # Run model-config-tests pytests - this also generates checksums files
+          # Run model-config-tests pytests - this also generates checksums files.
+          # If there is a checksum to compare against, add --checksum-path arg.
           model-config-tests -s -m "${{ inputs.test-markers }}" \
             --output-path ${{ env.EXPERIMENT_LOCATION }} \
+            ${{ inputs.compared-config-ref != '' && '--checksum-path $COMPARED_CHECKSUM_FILE \' || '\' }}
```

Comment:
Sorry this may be a silly question as I don't know, but would `$COMPARED_CHECKSUM_FILE` be accessible in the expression `${{}}`? I'm just unsure on when the `${{}}` are evaluated if that makes sense and whether they have access to the current environments variables.

Reply:
Yeah, so the `${{ ... }}` is evaluated first outside the context of `bash`, as it kind of a value-substitution/templating thing that acts before anything in the workflow is run. After all the templating is done, the workflow is run, which means that the `$COMPARED_CHECKSUM_FILE` is interpreted correctly by `bash`. 
For example, expand the first step of https://github.com/ACCESS-NRI/access-om2-configs/actions/runs/9867364516/job/27247633735#step:3:1 and you will see that there is no `${{ ... }}`, only a normal looking bash script to be run. But in the workflow file there are a bunch of those template markers.

Reply:
Oh that makes total sense, thanks for explaining that!